runtime/common/src/fees.rs

```diff
@@ -116,63 +128,117 @@ macro_rules! impl_foreign_fees {
             Ok(converted_fee)
         }
 
+        fn get_fee_factor_campaign_asset(
+            campaign_asset: CampaignAsset,
+        ) -> Result<Balance, TransactionValidityError> {
+            let ztg_supply = Balances::total_issuance();
+            let campaign_asset_supply = AssetManager::total_issuance(campaign_asset.into());
+            let fee_multiplier = Balance::from(CampaignAssetFeeMultiplier::get());
+
+            // Use neutral fee multiplier if the ZTG supply is 100x greater than the campaign
+            // asset supply.
+            if ztg_supply.saturating_div(campaign_asset_supply) >= fee_multiplier {
+                Ok(BASE)
+            } else {
+                campaign_asset_supply.saturating_mul(fee_multiplier).bdiv(ztg_supply).map_err(
+                    |_| {
+                        TransactionValidityError::Invalid(InvalidTransaction::Custom(
+                            CustomTxError::FeeConversionArith as u8,
```

Comment:
`FeeConversionArith` perfectly describes what is going on from my perspective. What you are suggesting here implies that we would create an unique error for every place in the code where an error is potentially thrown, even if another error already perfectly describes what is happening. Not going down that path :grin:

Reply:
The reason I suggested this was that we once used the same error multiple times in the same function and therefore couldn't directly distinguish what has failed. In this case there are three possible places. But I think it's okay, since we can also distinguish where it was thrown ourselves to improve maintenance of errors on our side.